app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -141,7 +141,11 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
             buildInfo = GleanBuildInfo.buildInfo
         )
 
-        setStartupMetrics(components.core.store, settings())
+        // We avoid blocking the main thread on startup by setting startup metrics on the background thread.
+        val store = components.core.store
+        GlobalScope.launch(Dispatchers.Default) {
```

Comment:
nit: This should probably be the `IO` dispatcher since we're reading prefs from disk for this change.